disco_aws_automation/disco_subnet.py

```diff
@@ -0,0 +1,314 @@
+        self._tag_resource(subnet['SubnetId'])
+        logging.debug("%s subnet: %s", self.name, subnet)
+        return subnet
+
+    def _find_route_table_by_id(self, route_table_id):
+        params = dict()
+        params['RouteTableIds'] = [route_table_id]
+        try:
+            return handle_date_format(
+                self.boto3_ec2.describe_route_tables(**params)
+            )['RouteTables'][0]
+        except IndexError:
+            return None
+
+    def _find_route_table(self):
+        filters = copy.deepcopy(self._resource_filter)
+        filters['Filters'].append({'Name': 'tag:subnet', 'Values': [self.name]})
+        try:
+            return handle_date_format(
+                self.boto3_ec2.describe_route_tables(**filters)
+            )['RouteTables'][0]
+        except IndexError:
+            return None
+
+    def _create_route_table(self):
+        params = dict()
+        params['VpcId'] = self.metanetwork.vpc.vpc.id
+        route_table = handle_date_format(self.boto3_ec2.create_route_table(**params))['RouteTable']
+        self._tag_resource(route_table['RouteTableId'])
+        logging.debug("%s route table: %s", self.name, route_table)
+
+        return route_table
+
+    def _associate_route_table(self, route_table):
+        # Associate route table with subnet
+        params = dict()
+        params['SubnetId'] = self.subnet['SubnetId']
+        params['RouteTableId'] = route_table['RouteTableId']
+        self.boto3_ec2.associate_route_table(**params)
+
+    def _create_and_associate_route_table(self):
+        route_table = self._create_route_table()
+        self._associate_route_table(route_table)
+        return route_table
+
+    def _find_nat_gateway(self):
+        params = dict()
+        params['Filters'] = []
+        params['Filters'].append({'Name': 'subnet-id', 'Values': [self.subnet['SubnetId']]})
+        params['Filters'].append({'Name': 'vpc-id', 'Values': [self.metanetwork.vpc.vpc.id]})
+        try:
+            result = handle_date_format(
+                self.boto3_ec2.describe_nat_gateways(**params)
+            )['NatGateways'][0]
+        except IndexError:
+            return None
+
+        if result['NatGatewayAddresses'][0]['AllocationId'] != self.nat_eip_allocation_id:
+            raise RuntimeError("EIP allocation id ({0}) doesn't match with existing "
+                               "NAT gateway's allocation id ({1}) in subnet ({2})."
+                               .format(self.nat_eip_allocation_id,
+                                       result['NatGatewayAddresses'][0]['AllocationId'],
+                                       self.subnet['SubnetId']))
+
+        return result
+
+    def _create_nat_gateway(self):
+        params = dict()
+        params['SubnetId'] = self.subnet['SubnetId']
+        params['AllocationId'] = self.nat_eip_allocation_id
+        logging.debug("Creating NAT gateway: %s", self.nat_eip_allocation_id)
```

Comment:
Somewhat preferable to make this log on a package-specific (or object-specific, which would have the same effect) logger—makes it easier for users to tune what debug statements they see.